pkg/datapath/garp/events.go

```diff
@@ -0,0 +1,180 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package garp
+
+import (
+	"context"
+	"net/netip"
+
+	"github.com/cilium/workerpool"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	nodeTypes "github.com/cilium/cilium/pkg/node/types"
+)
+
+var nodePodsCell = cell.Provide(func(lc hive.Lifecycle, c k8sClient.Clientset) (resource.Resource[*corev1.Pod], error) {
+	if !c.IsEnabled() {
+		return nil, nil
+	}
+
+	lw := utils.ListerWatcherWithFields(utils.ListerWatcherFromTyped[*corev1.PodList](
+		c.CoreV1().Pods("")), fields.OneTermEqualSelector("spec.nodeName", nodeTypes.GetName()),
+	)
+	return resource.New[*corev1.Pod](lc, lw), nil
+})
+
+// EventProcessorCell processes k8s pod events for the current node and
+// determines if a Gratuitous ARP packet needs to be sent.
+var EventProcessorCell = cell.Module(
```

Comment:
Consider putting this into `cell.go` and naming it `var Cell`. Move `config.go` contents into it as well and merge `Cell` from `garp.go` into it. Idea being that `cell.go` defines the "public" API of the module, e.g. the cell definition, exported interface types and the configuration.